internal/worker/concurrency.go

```diff
@@ -18,21 +18,28 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package autoscaler
+package worker
 
-// AutoScaler collects data and estimate usage
-type (
-	AutoScaler interface {
-		Estimator
-		// Acquire X ResourceUnit of resource
-		Acquire(ResourceUnit) error
-		// Release X ResourceUnit of resource
-		Release(ResourceUnit)
-		// GetCurrent ResourceUnit of resource
-		GetCurrent() ResourceUnit
-		// Start starts the autoscaler go routine that scales the ResourceUnit according to Estimator
-		Start()
-		// Stop stops the autoscaler if started or do nothing if not yet started
-		Stop()
-	}
-)
+import "context"
+
+var _ Permit = (*resizablePermit)(nil)
+
+// ConcurrencyLimit contains synchronization primitives for dynamically controlling the concurrencies in workers
+type ConcurrencyLimit struct {
+	PollerPermit Permit // controls concurrency of pollers
+	TaskPermit   Permit // controls concurrency of task processing
+}
+
+// Permit is an adaptive permit issuer to control concurrency
+type Permit interface {
+	Acquire(context.Context) error
```

Comment:
do you mind adding comments for each method?

Reply:
sorry this branch was accidentally pushed. I'm still testing this branch https://github.com/cadence-workflow/cadence-go-client/pull/1411